Add NOSTDINC and NOTORCHINC lintrunner

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #6693
* #6692

Prohibit standard C++ container, aten, and torch includes in core, at least pending further review. Note that non-core directories, such as extension/, test/, and any testing_util/ subdirectories, are exempted.

Differential Revision: [D65541332](https://our.internmc.facebook.com/intern/diff/D65541332/)

swolchok committed:
[ExecuTorch] Add NOSTDINC and NOTORCHINC lintrunner

Prohibit standard C++ container, aten, and torch includes in core, at least pending further review. Note that non-core directories, such as extension/, test/, and any testing_util/ subdirectories, are exempted.

Differential Revision: [D65541332](https://our.internmc.facebook.com/intern/diff/D65541332/)

[ghstack-poisoned]

kimishpatel commented on .lintrunner.toml:
```diff
@@ -196,3 +196,87 @@ command = [
+    "**/*.h",
+    "**/*.hpp",
+]
+exclude_patterns = [
```
Instead of exclude, can you just do include for runtime directory?

swolchok replied:
> Instead of exclude, can you just do include for runtime directory?

directories should be linted for this unless we know that they shouldn't be. We want this to fail closed.

kimishpatel reviewed (changes requested):
Did you test that this works?

swolchok committed:
Update on "Add NOSTDINC and NOTORCHINC lintrunner"


Prohibit standard C++ container, aten, and torch includes in core, at least pending further review. Note that non-core directories, such as extension/, test/, and any testing_util/ subdirectories, are exempted.

Differential Revision: [D65541332](https://our.internmc.facebook.com/intern/diff/D65541332/)

[ghstack-poisoned]

swolchok commented:
> Did you test that this works?

yes see internal diff test plan

swolchok committed:
Update on "Add NOSTDINC and NOTORCHINC lintrunner"


Prohibit standard C++ container, aten, and torch includes in core, at least pending further review. Note that non-core directories, such as extension/, test/, and any testing_util/ subdirectories, are exempted.

Differential Revision: [D65541332](https://our.internmc.facebook.com/intern/diff/D65541332/)

[ghstack-poisoned]

swolchok committed:
Update on "Add NOSTDINC and NOTORCHINC lintrunner"


Prohibit standard C++ container, aten, and torch includes in core, at least pending further review. Note that non-core directories, such as extension/, test/, and any testing_util/ subdirectories, are exempted.

Differential Revision: [D65541332](https://our.internmc.facebook.com/intern/diff/D65541332/)

[ghstack-poisoned]
